### ISSUE TITLE
Interesting system

### ISSUE THREAD

**Issue by MostHated** [bzgeb/SerializableScriptableObject#2]
Hey there,
    I came across this and decided to check it out. I like the idea of using a scriptable to easily track each entity, at a quick glance, I thought it might have solved a pretty common issue of keeping track of scene items in an out of scene way, but when I tested it, it didn't quite do what I was expecting. 

It looks like the Guid is less about tracking the individual entities in a unique way, but more about keeping track of them as a whole?
I tried to take a look at the ID's that were being saved and saw that they ended up being all the same for each of the types.

![](https://i.imgur.com/lCYc5Ax.png)

I still like the idea and overall system though, I can definitely see how it would be useful in many situations. :+1: 
Have a good one,
-MH

**Comment by bzgeb** [comment]
Yeah that's right. If you want to track individual entities inside a scene, you can your own GUID (as in one not generated by Unity) and serialize that along with the scene. 

[This code](https://gist.github.com/bzgeb/8c389cfd8c59be262cc7071d25990ffe) might be helpful as a starting point. 

It depends what you're trying to accomplish, but basically you can use that code to add a UUID field to any entities you want to track. Then, generate a new UUID for each entity in the scene and it'll be saved alongside the scene.

Hope that helps!

**Comment by MostHated** [comment]
I appreciate the example. I already came up with my own system which is quite similar to what you posted a while back that I have been using. I just have a habit of checking out newer repos like this to see if perhaps something was figured out that I might not have thought of, or could learn something new from. 

I have tried experimenting with all kinds of different ways to try and track/reference objects, but there is always some caveat that can't be overcome without the system becoming so complex that it ends up not even being worth using anymore. I tried going about it in a similar manner to how Timeline Director keeps track of objects (https://github.com/instance-id/SO-Persistent-Reference), while it was a good learning experience figuring it out and putting that together, it still wasn't quite what I was ultimately after. I think the only way I will truly get solid, reliable object tracking is for Unity to implement it on the C++ side, but I don't see that happening anytime soon, unfortunately.
